packages/design-system/src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,436 @@
+  /**
+   * Buttons or other HTML to be rendered in the "actions" bar
+   * at the bottom of the dialog.
+   */
+  actions?: React.ReactNode;
+  /**
+   * Additional classes to be added to the actions container.
+   */
+  actionsClassName?: string;
+  /**
+   * Aria label for the close button
+   */
+  ariaCloseLabel?: string;
+  /**
+   * Additional classes to be added to the root dialog element.
+   */
+  className?: string;
+  /**
+   * Size of the close button. See [Button component]({{root}}/components/button/#components.button.react)
+   */
+  closeButtonSize?: DialogCloseButtonSize;
+  /**
+   * For internationalization purposes, the text for the "Close" button must be
+   * passed in as a prop.
+   */
+  closeButtonText?: React.ReactNode;
+  /**
+   * Variation string to be applied to close button component. See [Button component]({{root}}/components/button/#components.button.react)
+   */
+  closeButtonVariation?: ButtonVariation;
+  /**
+   * The icon to display as part of the close button
+   */
+  closeIcon?: React.ReactNode;
+  /**
+   * @hide-prop [Deprecated] This prop has been renamed to `closeButtonText`
+   * @hide-prop The text for the "Close" button
+   */
+  closeText?: React.ReactNode;
+  /**
+   * @hide-prop [Deprecated] This prop has been renamed to `escapeExits`.
+   * @hide-prop Disable exiting the dialog when a user presses the Escape key.
+   */
+  escapeExitDisabled?: boolean;
+  /**
+   * Additional classes to be added to the header, which wraps the heading and
+   * close button.
+   */
+  headerClassName?: string;
+  /**
+   * The Dialog's heading, to be rendered in the header alongside the close button.
+   */
+  heading?: React.ReactNode;
+  /**
+   * The Dialog's size parameter.
+   */
+  size?: DialogSize;
+  /**
+   * @hide-prop [Deprecated] This prop has been renamed to `heading`.
+   */
+  title?: React.ReactNode;
+  /**
+   * Allow additional AriaModal props to be passed to Dialog
+   */
+  // eslint-disable-next-line camelcase
+  additional_props?: AriaModalProps;
+}
+
+export const Dialog = (props: DialogProps): React.ReactNode => {
+  const headingRef = useRef(null);
+  let eventHeadingText = '';
+
+  if (process.env.NODE_ENV !== 'production') {
+    if (props.title) {
+      console.warn(
+        `[Deprecated]: Please remove the 'title' prop in <Dialog>, use 'heading' instead. This prop has been renamed and will be removed in a future release.`
+      );
+    }
+    if (props.escapeExitDisabled) {
+      console.warn(
+        `[Deprecated]: Please remove the 'escapeExitDisabled' prop in <Dialog>, use 'escapeExits' instead. This prop has been renamed and will be removed in a future release.`
+      );
+    }
+    if (props.closeText) {
+      console.warn(
+        `[Deprecated]: Please remove the 'closeText' prop in <Dialog>, use 'closeButtonText' instead. This prop has been renamed and will be removed in a future release.`
+      );
+    }
+  }
+
+  useEffect(() => {
```

Comment:
This is something I'm noticing on my HelpDrawer conversion... but refactoring the lifecycle methods used for analytics with `useEffect` breaks the analytics unit tests. I don't understand why this is the case though and can only get my own tests to pass by maintaining the class component structure.

I wonder if it has something to do with a class property in the constructor behaving different from a func let? It's fuzzy to me.

Reply:
the analytics issue is a race condition in the test for the analytics being ready in the event queue, adding a `setTimeout(() => { test },0)` sorts it out, updated the test, works now!